metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala

```diff
@@ -256,19 +271,24 @@ final class FileDecoderProvider(
 
   private def findClassesDirFileFromSource(
       sourceFile: AbsolutePath,
-      newExtension: String
-  ): Option[PathInfo] = {
+      newExtension: String,
+      position: l.Position = new l.Position(1, 1)
+  ): Option[PathInfo] =
     for {
       targetId <- buildTargets.sourceBuildTargets(sourceFile).headOption
       target <- buildTargets.scalaTarget(targetId)
       sourceRoot <- buildTargets.inverseSourceItem(sourceFile)
-      classDir = target.classDirectory.toAbsolutePath
-      oldExtension = sourceFile.extension
-      relativePath = sourceFile
-        .toRelative(sourceRoot)
-        .resolveSibling(_.stripSuffix(oldExtension) + newExtension)
-    } yield PathInfo(Some(targetId), classDir.resolve(relativePath))
-  }
+      pathToClass <- classFinder.resolveClassName(
```

Comment:
Shouldn't this be a different logic for tasty files and normal classfiles? We can fix tasty first and work on the other on a different PR.

Reply:
But resolving logic is almost the same, it only differs when it comes to inner classes.

Reply:
Sure, but is there any way we check if it's tasty or classfile?

Reply:
For now, tasty and class files have different logic. Maybe asking the user to pick which class he wants to decompile would solve this problem at all.